### DISCUSSION TITLE
Rosetta crashed while designing with PDB.rotamer

### DISCUSSION THREAD

**Discussion by MuhammadAttaElhamouly** [RosettaCommons/rosetta#141] [Rosetta] [answered]
Hi,

I tried out "Non-canonical amino acid (NCAA) parameterization using Rosetta" from the Meiler Lab 2023 tutorial, I used the fake_rotlib script to make libraries for some amino acids, each of 2000 rotamers (as the structure is large), and when I moved to the next part of the tutorial "Macrocyclic peptide design in Rosetta" which relies on research article of @vmullig  Mulligan et. al. Computationally designed peptide macrocycle inhibitors of New Delhi metallo-beta-lactamase 1, the submitted job crashed after rising of the occupied size RAM (250 GB of the HPC), without any details about why that happened.

Note, that using only one or 2 amino acids worked without issues, only more numbers resulted in that.

**Comment by vmullig** [comment, marked as answer]
2000 rotamers is a huge rotamer library.  It could easily result in you
running out of memory at any of the following steps:

- Read-in of the rotamer library, if the dimensionality is large.
- Setup of the polycubic interpolation used for scoring.
- Construction of the list of candidate rotamer instances for a particular
packing problem.
- Precomputation of the interaction graph, which needs to store pairwise
energies for all of the potentially interacting rotamers (so if you had
just two positions each with just 2000 rotamers, this would be 4 million
pairwise interactions).
- Precomputation of other quantities stored by design-centric guidance
terms (voids_penalty, hbnet, buried_unsatisfied_penalty).

Try running on a single core without the -mute all flag to see what Rosetta
was trying to do when it ran out of memory.

**Reply by MuhammadAttaElhamouly** [reply]
Thanks, I'll check it out again according to your recommendations.

**Comment by vmullig** [comment]
For more verbose output, you can also add:

-out:level 500
